integration/tests/basic/test_batch_operations.py

```diff
@@ -0,0 +1,162 @@
+
+            batch.append(
+                {
+                    "id": id_,
+                    "jsonrpc": "2.0",
+                    "method": "eth_call",
+                    "params": [
+                        {
+                            "from": account.address,
+                            "to": common_contract.address, data_or_input_field_name: data,
+                        },
+                        "latest",
+                    ],
+                }
+            )
+
+        response = requests.post(
+            url=request.config.environment.proxy_url,  # noqa
+            json=batch,
+        )
+
+        results = response.json()
+        msg = f"Sent batch length {len(expected_results)}, but received {len(results)} results"
+        assert len(results) == len(expected_results), msg
+
+        for index, result in enumerate(results):
+            assert "error" not in result
+            assert result["id"] == index + 1, f"Batch response response sequence invalid"
+
+            result_hex = result["result"]
+            actual_function_return = int(result_hex, 16)
+            expected_function_return = expected_results[index]
+            assert actual_function_return == expected_function_return, "Invalid function return"
+
+    def test_batch_operations_different_functions(
+            self,
+            request: pytest.FixtureRequest,
+            common_contract,
+            json_rpc_client: JsonRPCSession,
+    ):
+        """
+        sends a batch operation request with a few calls to different functions
+        """
+        sender_account = self.accounts[0]
+        recipient_account = self.accounts[1]
+        transaction = self.web3_client.make_raw_tx(
+            from_=sender_account, to=recipient_account, amount=1, estimate_gas=True
+        )
+        signed_tx = self.web3_client.eth.account.sign_transaction(transaction, sender_account.key)
+
+        batch = [
+            {
+                "jsonrpc": "2.0",
+                "method": "eth_chainId",
+                "params": [],
+                "id": 1,
+            },
+            {
+                "jsonrpc": "2.0",
+                "method": "eth_gasPrice",
+                "params": [],
+                "id": 2,
+            },
+            {
+                "jsonrpc": "2.0",
+                "method": "eth_sendRawTransaction",
+                "params": [signed_tx.rawTransaction.hex()],
+                "id": 3,
+            },
+        ]
+
+        response = requests.post(
+            url=request.config.environment.proxy_url,  # noqa
+            json=batch,
+        )
+
+        results = response.json()
+        msg = f"Sent batch length {len(batch)}, but received {len(results)} results"
+        assert len(results) == len(batch), msg
+
+        for result in results:
+            assert "error" not in result
+            assert rpc_checks.is_hex(result["result"])
+
+    def test_batch_operations_negative(
```

Comment:
I suggest to add one case with all positive and 1 negative requests somewhere in the middle
And some request wich return error from a contract ( not bad parameters error)